### PR TITLE
[top] Add testplan item for RMA bootstrap

### PR DESCRIPTION
**Pull request by tjaychen**
- Tests the rom logic for rma bootstrap.
- Test contains two iterations, one positive test and one negative test.

Signed-off-by: Timothy Chen <timothytim@google.com>

### CODE REVIEW

**Review by moidx — APPROVED** [review]

File: hw/top_earlgrey/data/chip_testplan.hjson [hunk 1]
```diff
@@ -3292,5 +3292,25 @@
...
+      - Drive software strap to the RMA bootstrap value.
+      - The test contains two iterations.
+      - In iteration 1, do not issue the life cycle rma command. ROM should timeout on spin cycles
+        and automatically reset the device. Check to ensure device has reset via backdoor CSR.
+      - In iteration 2, issue the life cycle RMA command and ensure RMA transition can be successfully
+        completed.
+      '''
+      stage: V2
+      tests: []
+    }
```
Comment (sriyerg) [inline hw/top_earlgrey/data/chip_testplan.hjson:3314]: Should we move this to the ROM testplan?

Comment (tjaychen) [reply hw/top_earlgrey/data/chip_testplan.hjson:3314]: ah yeah good point. You mean chip_rom_testplan right?

**Review by alphan — APPROVED** [review]
Thanks @tjaychen! Just to make sure that we are on the same page: I plan to add a e2e test for cw310 for "iteration 1". DV obviously can cover both cases.

**Comment by alphan** [comment]
@tjaychen I can create an issue for this test in the rom-e2e project using my script once this is merged (feel free to skip creating now for now).

**Comment by tjaychen** [comment]
o sounds good! do you want me to update the tags?

File: sw/device/silicon_creator/rom/data/rom_e2e_testplan.hjson [hunk 2]
```diff
@@ -1157,5 +1157,26 @@
...
+      desc: '''End to end test to ensure RMA mode can be activated by halting ROM execution.
+
+      - This test must be run with the real ROM.
+      - Pre-load the device into PROD or PROD_END state.
+      - Backdoor load CREATOR_SW_CONFIG_RMA_SPIN_CYCLES and SPIN_EN.
+        - The value for CREATOR_SW_CONFIG_RMA_SPIN_CYCLES should be small, just enough
+          for the DV agent to activate the life cycle JTAG interface.
+      - Drive software strap to the RMA bootstrap value.
+      - The test contains two iterations.
+      - In iteration 1, do not issue the life cycle rma command. ROM should timeout on spin cycles
```
Comment (msfschaffner) [inline sw/device/silicon_creator/rom/data/rom_e2e_testplan.hjson:1172]: nit: rma -> RMA

**Review by msfschaffner — APPROVED** [review]
Thanks @tjaychen!

**Pull request merged**